src/core/src/core_logic/PatchInstaller.py

```diff
@@ -326,6 +335,13 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
 
         stopwatch_for_sequential_install_process.stop_and_write_telemetry(sequential_processing_perf_log)
 
+        if not patch_installation_successful or maintenance_window_exceeded:
+            message = "\n\nOperation status was marked as failed because: "
+            message += "[X] a failure occurred during the operation  " if not patch_installation_successful else ""
+            message += "[X] maintenance window exceeded " if maintenance_window_exceeded else ""
+            self.status_handler.add_error_to_status(message, Constants.PatchOperationErrorCodes.OPERATION_FAILED)
+            self.composite_logger.log_error(message)
+
```

Comment:
log_final_metrics() has 2 callers line 324 and line 217, moving this if block here would work for line 324 but not 217.

Reply:
Did not get it. @rane-rajasi / @GAURAVRAMRAKHYANI : I could test the L217 with this change. What is the ask here?

Reply:
Why is if block moved here at line 338 from log_final_metrics() method?

There are some useful logging in this if block which should get executed when log_final_metrics() is called. log_final_metrics() has currently 2 callers.

Reply:
Got it . I think it was merge issue. Will change it.

Reply:
Done.